Lib/_collections_abc.py

```diff
@@ -899,6 +899,10 @@ def __reversed__(self):
     def index(self, value, start=0, stop=None):
         '''S.index(value, [start, [stop]]) -> integer -- return first index of value.
            Raises ValueError if the value is not present.
+
+           It is not required that all concrete implementations of
+           the Sequence class support start and stop attributes.
```

Comment:
I would rephrase it as:

Supporting start and stop optional arguments is optional, but recommended.

Reply:
I changed it to 'Supporting start and stop arguments is optional, but recommended.". I removed the first 'optional' word because the senses of both 'optional' words is different and might confuse the reader and also the sense of first 'optional' is conveyed by the function signature anyway.